collectors/0/grok_exporter.py

```diff
@@ -113,62 +116,55 @@ def kill_children(signum):
                 for pattern in patterns:
                     matcher = pattern.search(line)
                     if matcher is not None:  # TODO would checking for startswith be more efficient here?
-#			print('Pattern matched on ' + line)
-                        found = True
-#			print (found)
                         g = matcher.groups()
-			if g[1].startswith('_count'):
-				continue
-			elif g[1].startswith('_sum'):
-				continue
-#			print(g)
+                        if g[1].startswith('_count'):
+                            continue
+                        elif g[1].startswith('_sum'):
+                            continue
                         tags = ' '.join(g[1].replace('"', '').strip('{}').split(','))
- 			if '::_' in g[0]:
-#				print('Found pattern')
-				extratags = g[0].split('::_');
-#				print(extratags)
-#				g[0] = extratags[0]
-#				print(g[0])
-#				print('Here')
-				global finalmetricname
-				finalmetricname = extratags[0]
-#				print(finalmetricname)
-				for i, extratag in enumerate(extratags):
-					if i == 0:
-						continue;
-					elif extratag.startswith('_'):
-						finalmetricname += extratag
-					else:
-						tags += ' ' + extratag.replace('_', '=')
-				buf = '%s %s%s %s' % (finalmetricname, timestamp, g[2], tags)
-			else:
-	                        buf = '%s %s%s %s' % (g[0], timestamp, g[2], tags)
-                        sys.stdout.write(buf)
-                        sys.stdout.write('\n')
-            if not found:
-                die()
-            sys.stdout.flush()
-#	except UnboundLocalError as err:
-#	    print("Type error: {0}".format(err))
-#	except TypeError as err:
-#	    print("Type error: {0}".format(err))
+                        if '::_' in g[0]:
+                            extratags = g[0].split('::_');
+                            global finalmetricname
+                            finalmetricname = extratags[0]
+                            for i, extratag in enumerate(extratags):
+                                if i == 0:
+                                    continue;
+                                elif extratag.startswith('_'):
+                                    finalmetricname += extratag
+                                else:
+                                    tags += ' ' + extratag.replace('_', '=')
+                            print_metric(finalmetricname, timestamp, g[2], tags)
+                            metric_naming.print_if_apptuit_standard_metric(finalmetricname, METRIC_MAPPING, timestamp, format_metric_value(g[2]),
+                                                                           tags=None, tags_str=tags)
+                        else:
+                            print_metric(g[0], timestamp, g[2], tags)
+                            metric_naming.print_if_apptuit_standard_metric(g[0], METRIC_MAPPING, timestamp, format_metric_value(g[2]), tags=None,
+                                                                           tags_str=tags)
```

Comment:
If not found then die should it be reinstated? What are the issues? Diff with old grok exporter and other use cases.

Reply:
Ignore this. This needs grokker process health metrics separately.